app/views/questionnaire.py

```diff
@@ -280,10 +301,21 @@ def post_block_handler(
             return redirect(next_url)
 
     questionnaire_store_updater.save_answers(form)
-    next_location = path_finder.get_next_location(current_location=current_location)
 
-    if _is_end_of_questionnaire(block, next_location):
-        return submit_answers(routing_path, schema)
+    if _user_posted_last_block_on_section_routing_path(
+        block_id, questionnaire_store.completed_items
+    ):
+        current_section = schema.get_section_via_block_id(block_id)
+        questionnaire_store_updater.add_completed_section(current_section['id'])
```

Comment:
This logic should be done in the questionnaire store updater when calling `save_answers`.